### DISCUSSION TITLE
New flask-migrate Error - Can't Find Other Instances of It

### DISCUSSION THREAD

**Discussion by daviddirico** [miguelgrinberg/Flask-Migrate#496] [Q&A] [answered]
I recently made some changes to the directories in my flask project and aligned to a more appropriate "__init__.py" file to store my app initializations.  (Before, I had everything, including routes in a single file).  That works splendidly now...

However, I haven't even made changes to my database models since doing this, and I casually ran a `flask migrate - m "foobar"`, and instead of flask returning with a safe message like "no changes detected" it pumped out this error.

```
  File "migrations/env.py", line 87, in run_migrations_online
  **current_app.extensions['migrate'].configure_args
TypeError: configure() got multiple values for keyword argument 'compare_type'
```

Looking at the file, the code in question has a HARD CODED value for `compare_type=True`
```
    with connectable.connect() as connection:
        context.configure(
            connection=connection,
            target_metadata=target_metadata,
            compare_type=True,
            process_revision_directives=process_revision_directives,
            **current_app.extensions['migrate'].configure_args
        )
```

I have not seen anyone ever get this error before based on some thorough searches.  And I don't know how this file works exactly.  But what could be the issue here?

I am close to uninstalling flask-migate and having to scrape everything together manually again, but I really do not want to!

**Comment by miguelgrinberg** [comment, marked as answer]
@daviddirico my guess is that you hand edited this file at some point, maybe long ago, to add the `configure_type` option. I assume you just upgraded to the latest Flask-Migrate release, which actually sets this option to `true` by default, so you can remove that line from your env.py file and let Flask-Migrate handle the option.

**Reply by daviddirico** [reply]
@miguelgrinberg That 100% describes what I did with my versioning -- and upon removing that argument, it defaulted to the expected input -- I am all cleared up.

Also, I started my application based on your React+Flask & Flask Mega tutorials and I have been able to learn so much because of your teachings.

Thank you!!
